gobblin-temporal/src/main/java/org/apache/gobblin/temporal/ddm/activity/impl/GenerateWorkUnitsImpl.java

```diff
@@ -73,13 +77,22 @@ public int generateWorkUnits(Properties jobProps, EventSubmitterContext eventSub
       // before embarking on (potentially expensive) WU creation, first pre-check that the FS is available
       FileSystem fs = JobStateUtils.openFileSystem(jobState);
       fs.mkdirs(workDirRoot);
-
-      List<WorkUnit> workUnits = generateWorkUnitsForJobState(jobState, eventSubmitterContext, closer);
-
+      boolean canCleanUpTempDirs = false; // unlike `AbstractJobLauncher` running the job end-to-end, this is Work Discovery only, so WAY TOO SOON for cleanup
+      DestinationDatasetHandlerService datasetHandlerService = closer.register(
+          new DestinationDatasetHandlerService(jobState, canCleanUpTempDirs, eventSubmitterContext.create()));
```

Comment:
why relocate from `generateWorkUnitsFromJobState`?

Reply:
Ah its because my first iteration of this had a refactor of the dataset handler service which I reverted after some thought, will change back